src/charm.py

```diff
@@ -372,35 +384,76 @@ def _render_template(self, filename, **kwargs):
         template = self.jinja2_env.get_template(filename)
         return template.render(**kwargs)
 
+    def _remove_cilium_vxlan(self) -> None:
+        ip = IPRoute()
+        try:
+            idx = ip.link_lookup(ifname="cilium_vxlan")
+            if idx:
+                ip.link("del", index=idx[0])
+        except Exception as e:
+            print(f"Error in removing the cilium interface: {e}")
+        finally:
+            ip.close()
```

Comment:
Upstream uses a context manager to handle the lifecycle of the IPRoute. What if we refactor it?
```suggestion
    def _remove_cilium_vxlan(self) -> None:
        with IPRoute() as ip:
            try:
                idx = ip.link_lookup(ifname="cilium_vxlan")
                if len(idx) > 0:
                    ip.link("del", index=idx[0])
            except Exception as e:
                log.exception("Error in removing the cilium interface")
```
Note: I've added a few changes to the implementation (logs/length checks)

Reply:
Nice! Thank you, Done